lib/utils/usage_docs.go

```diff
@@ -0,0 +1,299 @@
+		}
+	}
+	return false
+}
+
+// argsToColumns outputs data for a table that lists arguments, their default
+// values, and help texts.
+func argsToColumns(a []*kingpin.ArgModel) [][3]string {
+	rows := [][3]string{}
+	for _, arg := range a {
+		if arg.Hidden {
+			continue
+		}
+
+		// Some commands declare empty argument names and help texts as
+		// a hack to allow arbitrary values. Indicate this in the table
+		// as a special case.
+		var argName string
+		if arg.Name != "" {
+			argName = arg.Name
+		} else {
+			argName = "args"
+		}
+
+		var help string
+		if arg.Help != "" {
+			help = formatHelp(arg.Help)
+		} else {
+			help = "Arbitrary arguments"
+		}
+
+		rows = append(rows, [3]string{
+			argName,
+			formatDefaultArgValue(arg),
+			help,
+		})
+	}
+	return rows
+}
+
+// envVarsToColumns prints table data for a list of environment variables, their
+// default values, and help texts.
+func envVarsToColumns(args []*kingpin.ArgModel, flags []*kingpin.FlagModel) [][3]string {
+	rows := [][3]string{}
+	for _, arg := range args {
+		if arg.Hidden || arg.Envar == "" {
+			continue
+		}
+
+		rows = append(rows, [3]string{
+			fmt.Sprintf("`%v`", arg.Envar),
+			formatDefaultArgValue(arg),
+			arg.Help,
+		})
+	}
+	for _, flg := range flags {
+		if flg.Hidden || flg.Envar == "" {
+			continue
+		}
+
+		rows = append(rows, [3]string{
+			fmt.Sprintf("`%v`", flg.Envar),
+			formatDefaultFlagValue(flg),
+			flg.Help,
+		})
+	}
+	return rows
+}
+
+// formatFlagForTable includes the names of flags so we can display them in a
+// table of flag information.
+func formatFlagForTable(haveShort bool, flag *kingpin.FlagModel) string {
+	flagString := ""
+	flagName := flag.Name
+	if flag.IsBoolFlag() {
+		flagName = "[no-]" + flagName
+	}
+	if flag.Short != 0 {
+		flagString += fmt.Sprintf("`-%c`, `--%s`", flag.Short, flagName)
+	} else {
+		if haveShort {
+			flagString += fmt.Sprintf("    `--%s`", flagName)
+		} else {
+			flagString += fmt.Sprintf("`--%s`", flagName)
+		}
+	}
+	return flagString
+}
+
+// cmdModelCollection is used to sort CmdModels for display in the CLI reference
+// docs page.
+type cmdModelCollection []*kingpin.CmdModel
+
+// Len returns the length of c. Required to implement sort.Interface.
+func (c cmdModelCollection) Len() int {
+	return len(c)
+}
+
+// Less returns whether command name i is lexicographically ordered before
+// command name j. Required to implement sort.Interface.
+func (c cmdModelCollection) Less(i, j int) bool {
+	return c[i].FullCommand < c[j].FullCommand
+}
+
+// Swap reverses the ordering of i and j within c. Required to implement
+// sort.Interface.
+func (c cmdModelCollection) Swap(i, j int) {
+	c[i], c[j] = c[j], c[i]
+}
+
+// sortcommandsByName sorts the commands in cmds by their full command names,
+// including all subcommands.
+func sortCommandsByName(cmds []*kingpin.CmdModel) []*kingpin.CmdModel {
+	col := cmdModelCollection(cmds)
+	sort.Stable(&col)
+	return []*kingpin.CmdModel(col)
+}
```

Comment:
I think you can probably replace this entire chunk with a single call to `slices.SortFunc`.